scrabble.js

```diff
@@ -1,14 +1,155 @@
 const Scrabble = {
+  values: {
+    A: 1, B: 3, C: 3, D:2, E: 1, F: 4, G: 2, H: 4, I: 1, J: 8, K: 5,
+    L: 1, M: 3, N: 1, O: 1, P: 3, Q: 10, R: 1, S: 1, T: 1, U: 1, V: 4,
+    W: 4, X: 8, Y: 4, Z: 10,
+  },
   score: function(word) {
-    // TODO: implement score
+    // TODO: implement score //score(word): returns the total score value for the given word. The word is input as a string (case insensitive). The chart below shows the point value for a given letter
+
+    let totalScore = 0;
+    let regex = /^[a-zA-Z]+$/;
+
+    word = word.toUpperCase();
+
+    if (regex.test(word) != true) {
+      throw 'Error';
+    }
+    // score adds 50 points for a 7-letter word
+    if (word.length == 7) {
+      totalScore += 50;
+    } else if (word.length > 7) {
+      throw 'Error';
+    }
+    // calculates score value for the given word
+    for (let letter of word) {
+      totalScore += this.values[letter];
+    }
+    return totalScore;
+  },
+
+  highestScoreFrom: function(arrayOfWords) {
+    // returns the word in the array with the highest score
+    let scores = [];
+    // throw error if arrayOfWords is empty
+    if (arrayOfWords.length == 0) {
+      throw 'Error';
+    }
+    // score each word and push onto scores array
+    for (let word of arrayOfWords) {
+      wordScore = this.score(word);
+      scores.push(wordScore);
+    }
```

Comment:
This algorithm ends up being a little over-complicated. By my reading, your steps are:

1. Calculate a score for every word
1. Find those words that share the highest score
1. Use apply tie-breaking logic to those words

A simpler algorithm is to iterate through the list of words, keeping track of the best word / score seen so far. For each word:
- If the current word has a higher score than the best word, it's the new best word
- If the best word has a higher score, keep it
- If the two are tied, apply tie-breaking logic, favoring the existing best word.

This single-pass approach is more concise and (in my opinion) easier to understand.